streams/src/test/java/org/apache/kafka/streams/kstream/internals/KStreamImplTest.java

```diff
@@ -2330,4 +2348,266 @@ public void shouldNotAllowNullNamedOnProcessWithStores() {
+            final TestInputTopic<String, String> inputTopic =
+                driver.createInputTopic(input, Serdes.String().serializer(), Serdes.String().serializer());
+            final TestOutputTopic<String, String> outputTopic =
+                driver.createOutputTopic(output, Serdes.String().deserializer(), Serdes.String().deserializer());
+
+            inputTopic.pipeInput("A", "01", 5L);
+            inputTopic.pipeInput("B", "02", 100L);
+            inputTopic.pipeInput("C", "03", 0L);
+            inputTopic.pipeInput("D", "04", 0L);
+            inputTopic.pipeInput("A", "05", 10L);
+            inputTopic.pipeInput("A", "06", 8L);
+
+            final List<TestRecord<String, String>> outputExpectRecords = new ArrayList<>();
+            outputExpectRecords.add(new TestRecord<>("A", "01", Instant.ofEpochMilli(5L)));
+            outputExpectRecords.add(new TestRecord<>("B", "02", Instant.ofEpochMilli(100L)));
+            outputExpectRecords.add(new TestRecord<>("C", "03", Instant.ofEpochMilli(0L)));
+            outputExpectRecords.add(new TestRecord<>("D", "04", Instant.ofEpochMilli(0L)));
+            outputExpectRecords.add(new TestRecord<>("A", "05", Instant.ofEpochMilli(10L)));
+            outputExpectRecords.add(new TestRecord<>("A", "06", Instant.ofEpochMilli(8L)));
+
+            assertEquals(outputTopic.readRecordsToList(), outputExpectRecords);
+        }
+    }
+
+
+    @Test
+    public void shouldKTableFromKStreamCheckMaterializedStore() {
+        final Consumed<String, String> consumed = Consumed.with(Serdes.String(), Serdes.String())
+            .withName("table-process");
+
+        final StreamsBuilder builder = new StreamsBuilder();
+        final String storeName = "store";
+
+        final String input = "input";
+        builder.stream(input, consumed)
+            .toTable(Materialized.as(Stores.inMemoryKeyValueStore(storeName)));
+
+
+        try (final TopologyTestDriver driver = new TopologyTestDriver(builder.build(), props)) {
+            final TestInputTopic<String, String> inputTopic =
+                driver.createInputTopic(input, Serdes.String().serializer(), Serdes.String().serializer());
+            final KeyValueStore<String, String> store = driver.getKeyValueStore(storeName);
+
+            inputTopic.pipeInput("A", "01");
+            final Map<String, String> expectedStore = mkMap(mkEntry("A", "01"));
+
+            assertThat(asMap(store), is(expectedStore));
+        }
+    }
+
+    @Test
+    public void shouldKStreamToKTableForeignLeftJoin() {
+        final Consumed<String, String> consumed = Consumed.with(Serdes.String(), Serdes.String());
+        final StreamsBuilder builder = new StreamsBuilder();
+
+        final String input1 = "input1";
+        final String input2 = "input2";
+        final String output = "output";
+
+        final KTable<String, String> leftTable = builder.table(input1, consumed);
```

Comment:
We should create this table from a stream. Otherwise we don't verify that `toTable()` does materialize the table (as required if used in a foreign key join) automatically. (ie, `toTable()` should not take a `Materialized` parameter in this test, but the join must still work).